sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py

```diff
@@ -605,13 +612,13 @@ class ReceiveClient(AMQPClient):
     :type handle_max: int
     :param on_attach: A callback function to be run on receipt of an ATTACH frame.
      The function must take 4 arguments: source, target, properties and error.
-    :type on_attach: func[~uamqp.address.Source, ~uamqp.address.Target, dict, ~uamqp.errors.AMQPConnectionError]
+    :type on_attach: func[~pyamqp.endpoint.Source, ~pyamqp.endpoint.Target, dict, ~pyamqp.errors.AMQPConnectionError]
     :param encoding: The encoding to use for parameters supplied as strings.
      Default is 'UTF-8'
     :type encoding: str
     """
 
-    def __init__(self, hostname, source, auth=None, **kwargs):
+    def __init__(self, source, auth=None, **kwargs):
```

Comment:
I don't think source and hostname are always going to be the same?
For example, I'm not sure if hostname includes the entity, where as the source/target would.

Reply:
Yes, and we do parsing on the hostname in the SDK itself - source may/may not have that same formatting